sound/soc/sof/intel/hda.c

```diff
@@ -101,6 +106,14 @@ struct sdw_intel_ops sdw_callback = {
 static int sdw_ace2x_params_stream(struct device *dev,
 				   struct sdw_intel_stream_params_data *params_data)
 {
+	struct snd_soc_dai *d = params_data->dai;
+	struct snd_soc_dapm_widget *w = snd_soc_dai_get_widget(d, params_data->substream->stream);
+
+	if (!w) {
+		dev_err(dev, "%s: widget not found, check amp link num in the topology\n",
+			__func__);
+		return -EINVAL;
+	}
```

Comment:
@bardliao, the `w` is unused here, do we really need this check added to this specific function?

Reply:
Checking here is to be consistent as sdw_params_stream(). Both sdw_params_stream() and sdw_ace2x_params_stream() are .params_stream callback. We can also check in sdw_hda_dai_hw_params(). What do you prefer?

Reply:
imho you should check for NULL pointer where it can cause issues. You will catch this problem anyways, but if you insist of doing it here, then do a simple
```c
if (!snd_soc_dai_get_widget(params_data->dai, params_data->substream->stream)) {
}
```
perhaps as forward looking check, but again, in this path the `w` is not used, so why check?

Reply:
I see, so yes, you should check in `sdw_hda_dai_hw_params()`, not here.

Reply:
Updated, thanks.